<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SettingsTab</class>
 <widget class="QWidget" name="SettingsTab">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>523</width>
    <height>388</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_4">
   <property name="spacing">
    <number>0</number>
   </property>
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QScrollArea" name="scrollArea">
     <property name="frameShape">
      <enum>QFrame::NoFrame</enum>
     </property>
     <property name="widgetResizable">
      <bool>true</bool>
     </property>
     <widget class="QWidget" name="scrollAreaWidgetContents">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>506</width>
        <height>789</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_5">
       <item>
        <widget class="QGroupBox" name="confGrpBox">
         <property name="title">
          <string>Config File</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout">
            <item>
             <widget class="QLabel" name="label">
              <property name="text">
               <string>Current Path:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="Conf_clearButton">
              <property name="text">
               <string>Clear</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="Conf_importButton">
              <property name="text">
               <string>Import</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="Conf_exportButton">
              <property name="text">
               <string>Export</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>0</width>
                <height>0</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QLabel" name="Conf_currPathLabel">
            <property name="text">
             <string notr="true">Path</string>
            </property>
            <property name="textInteractionFlags">
             <set>Qt::LinksAccessibleByMouse|Qt::TextSelectableByMouse</set>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="Conf_createInCWDButton">
            <property name="text">
             <string>Create one in working directory</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="Conf_createInConfDirButton">
            <property name="text">
             <string>Create one in config directory</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPlainTextEdit" name="Conf_tipsEdit">
            <property name="undoRedoEnabled">
             <bool>false</bool>
            </property>
            <property name="readOnly">
             <bool>true</bool>
            </property>
            <property name="plainText">
             <string>You can specify the config file path by passing &quot;--config-path &lt;path&gt;&quot; option when opening it.
If the config file path is not set, this app will use the config file in the working directory first.
If the file is not in the working directory, the app will will find it in the configuration directory.
If no config file is detected, This app will create one in the current working directory.</string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_4">
            <item>
             <widget class="QLabel" name="label_6">
              <property name="text">
               <string>Max History Count *:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="Conf_maxHistoryBox">
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>2147483640</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="Conf_setMaxHistoryButton">
              <property name="text">
               <string>Set</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="Conf_clearHistoryButton">
              <property name="text">
               <string>Clear History</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer_3">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>0</width>
                <height>0</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="langGrpBox">
         <property name="title">
          <string>Language *</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <widget class="QComboBox" name="Lang_nameBox">
              <property name="sizeAdjustPolicy">
               <enum>QComboBox::AdjustToContents</enum>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="Lang_setButton">
              <property name="text">
               <string>Set</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer_2">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>0</width>
                <height>0</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QLineEdit" name="Lang_filePathEdit">
            <property name="placeholderText">
             <string>Language file path (*.qm)</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="appearanceGrpBox">
         <property name="title">
          <string>Appearance</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <widget class="QCheckBox" name="Android_fullScreenBox">
            <property name="text">
             <string>Full Screen</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="Android_forceLandscapeBox">
            <property name="text">
             <string>Force Landscape Mode</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="Android_dockBox">
            <property name="text">
             <string>Use Dock * (Experimental)</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QWidget" name="opacityWidget" native="true">
            <layout class="QHBoxLayout" name="horizontalLayout_3">
             <property name="leftMargin">
              <number>0</number>
             </property>
             <property name="topMargin">
              <number>0</number>
             </property>
             <property name="rightMargin">
              <number>0</number>
             </property>
             <property name="bottomMargin">
              <number>0</number>
             </property>
             <item>
              <widget class="QLabel" name="label_2">
               <property name="text">
                <string>Opacity:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSlider" name="Opacity_slider">
               <property name="minimum">
                <number>10</number>
               </property>
               <property name="maximum">
                <number>100</number>
               </property>
               <property name="value">
                <number>100</number>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="tickPosition">
                <enum>QSlider::TicksBelow</enum>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="Opacity_Box">
               <property name="minimum">
                <number>10</number>
               </property>
               <property name="maximum">
                <number>100</number>
               </property>
               <property name="value">
                <number>100</number>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_5">
            <item>
             <widget class="QLabel" name="label_9">
              <property name="text">
               <string>Theme:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="Theme_nameBox">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="Theme_setButton">
              <property name="text">
               <string>Set</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_7">
            <item>
             <widget class="QLabel" name="label_3">
              <property name="text">
               <string>Global Font:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QFontComboBox" name="Font_nameBox"/>
            </item>
            <item>
             <widget class="QSpinBox" name="Font_sizeBox">
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>65535</number>
              </property>
              <property name="value">
               <number>9</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="Font_setButton">
              <property name="text">
               <string>Set</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_8">
            <item>
             <widget class="QLabel" name="label_4">
              <property name="text">
               <string>Data Font:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QFontComboBox" name="DataFont_nameBox"/>
            </item>
            <item>
             <widget class="QSpinBox" name="DataFont_sizeBox">
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>65535</number>
              </property>
              <property name="value">
               <number>9</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="DataFont_setButton">
              <property name="text">
               <string>Set</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="generalGrpBox">
         <property name="title">
          <string/>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_6">
          <item>
           <widget class="QCheckBox" name="Android_HWSerialBox">
            <property name="text">
             <string>Enable Hardware Serial Port</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="Data_recordDataBox">
            <property name="text">
             <string>Remember data to be sent in DataTab</string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_11">
            <item>
             <widget class="QCheckBox" name="Data_mergeTimestampBox">
              <property name="text">
               <string>Merge Timestamps With Short Intervals:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="Data_mergeTimestampIntervalBox">
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>65535</number>
              </property>
              <property name="value">
               <number>10</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_13">
              <property name="text">
               <string>ms</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer_4">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QCheckBox" name="General_simultaneousClearBox">
            <property name="text">
             <string>Clear received data in DataTab and graph in PlotTab simultaneously</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="General_touchScrollBox">
            <property name="text">
             <string>Respond to Touch Scroll Events</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>Restart the app to make settings with &quot;*&quot; take effect.</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_10">
         <item>
          <widget class="QLabel" name="label_10">
           <property name="text">
            <string>Version:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="versionLabel">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_6">
         <item>
          <widget class="QLabel" name="label_7">
           <property name="text">
            <string>Latest Release:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_8">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string notr="true">&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;a href=&quot;https://github.com/wh201906/SerialTest/releases/&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#0000ff;&quot;&gt;https://github.com/wh201906/SerialTest/releases/&lt;/span&gt;&lt;/a&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="openExternalLinks">
            <bool>true</bool>
           </property>
           <property name="textInteractionFlags">
            <set>Qt::TextBrowserInteraction</set>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_9">
         <item>
          <widget class="QLabel" name="label_11">
           <property name="text">
            <string>Report Bugs:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_12">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string notr="true">&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;a href=&quot;https://github.com/wh201906/SerialTest/issues/&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#0000ff;&quot;&gt;https://github.com/wh201906/SerialTest/issues/&lt;/span&gt;&lt;/a&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="openExternalLinks">
            <bool>true</bool>
           </property>
           <property name="textInteractionFlags">
            <set>Qt::TextBrowserInteraction</set>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
